app/assets/javascripts/initializers/initNotifications.js

```diff
@@ -88,51 +86,55 @@ function fetchNotificationsCount() {
   }
 }
 
+function onClickReaction(event) {
+  event.preventDefault();
+  sendHapticMessage('medium');
+  var thisButt = this;
+  thisButt.classList.add('reacted');
+
+  function successCb(response) {
+    if (response.result === 'create') {
+      thisButt.classList.add('reacted');
+    } else {
+      thisButt.classList.remove('reacted');
+    }
+  }
+
+  var formData = new FormData();
+  formData.append('reactable_type', thisButt.dataset.reactableType);
+  formData.append('category', thisButt.dataset.category);
+  formData.append('reactable_id', thisButt.dataset.reactableId);
+
+  getCsrfToken()
+    .then(sendFetch('reaction-creation', formData))
+    .then(response => {
+      if (response.status === 200) {
+        response.json().then(successCb);
+      }
+    });
+}
+
 function initReactions() {
-  setTimeout(function() {
+  setTimeout(() => {
     if (document.getElementById('notifications-container')) {
       var butts = document.getElementsByClassName('reaction-button');
-      for (var i = 0; i < butts.length; i++) {
-        var butt = butts[i];
-        butt.onclick = function(event) {
-          event.preventDefault();
-          sendHapticMessage('medium');
-          var thisButt = this;
-          thisButt.classList.add('reacted');
-
-          function successCb(response) {
-            if (response.result === 'create') {
-              thisButt.classList.add('reacted');
-            } else {
-              thisButt.classList.remove('reacted');
-            }
-          }
-
-          var formData = new FormData();
-          formData.append('reactable_type', thisButt.dataset.reactableType);
-          formData.append('category', thisButt.dataset.category);
-          formData.append('reactable_id', thisButt.dataset.reactableId);
-
-          getCsrfToken()
-            .then(sendFetch('reaction-creation', formData))
-            .then(function(response) {
-              if (response.status === 200) {
-                response.json().then(successCb);
-              }
-            });
-        };
+      var i;
+      var butt;
+      for (i = 0; i < butts.length; i += 1) {
+        butt = butts[i];
+        butt.onclick = event => onClickReaction(event);
```

Comment:
here you have to pass `onClickReaction` directly:

```suggestion
butt.addEventListener('click', onClickReaction);
```

(event is also undefined)

let's also try in general to use `addEventListener` instead of the old `onclick` if possible

Reply:
Ok